legate/core/legate.py

```diff
@@ -80,7 +80,7 @@ def stores(self) -> list[Store]:
     @staticmethod
     def from_stores(
         dtype: pyarrow.DataType,
-        stores: list[Store],
+        stores: list[Optional[Store]],
```

Comment:
It's actually a bit surprising that you figured this out just by making mypy happy. was there any call site that required this type (which btw is correct)?

Reply:
Line 143 of `io.py` does this
```
  array = Array(dtype, [None, store])
```